### PR TITLE
Connect and restructured examples

### PR DESCRIPTION
**Pull request by andrewosh**
Added the connect function, which takes a pure component (a function of state), and a mapping from the global state to a set of properties, to produce a "smart" component. 

Also restructured the examples directory and added a few new packages.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
+var _ = require('lodash')
 var vdom = require('virtual-dom')
 var hyperx = require('hyperx')
 var hx = hyperx(vdom.h)
 
-var dx
+var dx, globalStore
+
+const filterState = function (mapping, state) {
```
Comment (freeman-lab) [inline index.js]: no const!

File: index.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
+var _ = require('lodash')
```
Comment (freeman-lab) [inline index.js]: can we just require the functions we really need, like `lodash.frompairs` and `lodash.get`?

File: package.json [hunk 3]
```diff
@@ -4,8 +4,10 @@
   "description": "really simple connector for redux",
   "main": "index.js",
   "scripts": {
-    "start": "budo example/index.js",
-    "test": "browserify test.js | testron | tap-spec"
+    "start": "budo examples/simple/index.js",
+    "test": "browserify test.js | testron | tap-spec",
+    "start:simple": "budo examples/simple/index.js",
+    "start:deep-components": "budo examples/deep-components/index.js"
```
Comment (freeman-lab) [inline package.json]: i would call deep-component -> connect, cause then i might just have three examples, display, dispatch, and connect

Comment (freeman-lab) [inline package.json]: if you make that name change now we can add the other two later

**Comment by andrewosh** [comment]
Made those changes!

**Comment by andrewosh** [comment]
Going to add a test case now

**Pull request merged**